ocaml/typing/typecore.ml

```diff
@@ -2729,8 +2747,7 @@ and type_pat_aux
       assert construction_not_used_in_counterexamples;
       (* Pretend separate = true *)
       let type_mode =
-        if has_local_attr_pat sp then Alloc_mode.Local
-        else Alloc_mode.Global
+        if explicitly_local then Alloc_mode.Local else Alloc_mode.Global
```

Comment:
Could we make the caller pass in `type_mode` instead of `explicitly_local`? One fewer layer of conceptual indirection.

Reply:
I decided to call it `explicit_mode`.